internal/controller/workspace/workspace.go

```diff
@@ -151,19 +157,41 @@ func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.E
 		return nil, errors.Wrap(err, errGetPC)
 	}
 
-	tf := c.terraform(dir)
 	switch cr.Spec.ForProvider.Source {
 	case v1alpha1.ModuleSourceRemote:
-		// TODO(negz): Handle errors initialising from a remote module.
-		// Terraform will refuse to init from a remote module if there
-		// are files in the workdir. On the other hand it sometimes
-		// needs these files, e.g. credentials or additional HCL. In the
-		// latter case it will download everything before failing so we
-		// work around the issue by attempting a best-effort init from
-		// a remote module, then adding the files, then later running a
-		// regular local init. The downside to this approach is that we
-		// currently drop errors indicating an invalid remote.
-		_ = tf.Init(ctx, terraform.FromModule(cr.Spec.ForProvider.Module))
+		// NOTE(ytsarev): Retrieve .git-credentials from Spec to /tmp outside of workspace directory
+		gitCredDir := filepath.Clean(filepath.Join("/tmp", dir))
+		if err := c.fs.MkdirAll(gitCredDir, 0700); err != nil {
+			return nil, errors.Wrap(err, errWriteGitCreds)
+		}
+		for _, cd := range pc.Spec.Credentials {
+			data, err := resource.CommonCredentialExtractor(ctx, cd.Source, c.kube, cd.CommonCredentialSelectors)
+			if err != nil {
+				return nil, errors.Wrap(err, errGetCreds)
+			}
+			if cd.Filename == ".git-credentials" {
+				p := filepath.Clean(filepath.Join(gitCredDir, filepath.Base(cd.Filename)))
+				if err := c.fs.WriteFile(p, data, 0600); err != nil {
+					return nil, errors.Wrap(err, errWriteGitCreds)
+				}
+			}
+		}
+		// NOTE(ytsarev): Make go-getter pick up .git-credentials, see /.gitconfig in the container image
+		err := os.Setenv("GIT_CRED_DIR", gitCredDir)
+		if err != nil {
+			return nil, errors.Wrap(err, errRemoteModule)
+		}
```

Comment:
Do we need to do this even when there's no `.git-credentials` file? Could it be moved into the `if cd.Filename == ".gitcredentials"` block?